vkcs/monitoring/resource_cloud_monitoring.go

```diff
@@ -0,0 +1,239 @@
+	capabilities, err := icapabilities.GetImageCapabilities(capabilitiesClient, imageID).Extract()
+	if err != nil {
+		resp.Diagnostics.AddError("Error when get VKCS capabilities client", err.Error())
+		return
+	}
+
+	if !isMonitoringSupported(&capabilities) {
+		resp.Diagnostics.AddError("Cloud monitoring is not supported on this image", "")
+		return
+	}
+
+	opts := monitoring.CreateUserOpts{
+		ImageID: imageID,
+		Capabilities: []string{
+			"telegraf",
+		},
+	}
+
+	settings, err := monitoring.CreateUser(monitoringClient, projectID, opts).Extract()
+	if err != nil {
+		resp.Diagnostics.AddError("Error when getting cloud monitoring settings", err.Error())
+		return
+	}
+
+	data.ID = types.StringValue(settings.UserID)
+	data.Region = types.StringValue(region)
+	data.Script = types.StringValue(settings.Script)
+	data.ServiceUserID = types.StringValue(settings.UserID)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, data)...)
+}
+
+func (r *сloudMonitoringResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data CloudMonitoringResourceModel
+
+	diags := req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *сloudMonitoringResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data *CloudMonitoringResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.AddError("Unable to update the cloud monitoring",
+		"Not implemented. Please report this issue to the provider developers.")
+}
+
+func (r *сloudMonitoringResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data CloudMonitoringResourceModel
+	diags := req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	region := data.Region.ValueString()
+	if region == "" {
+		region = r.config.GetRegion()
+	}
+
+	projectID := r.config.GetTenantID()
+
+	monitoringClient, err := r.config.CloudMonitoringV2Client(region, projectID)
+	if err != nil {
+		processMonitoringClientCreationErr(&resp.Diagnostics, err)
+		return
+	}
+
+	if data.ServiceUserID.IsNull() {
+		return
+	}
+
+	err = monitoring.DeleteUser(monitoringClient, projectID, data.ServiceUserID.ValueString()).Err
+	if err != nil {
+		if errutil.IsNotFound(err) {
+			return
+		}
+
+		resp.Diagnostics.AddError("Error when deleting cloud monitoring service user", err.Error())
+		return
+	}
+}
+
+func isMonitoringSupported(capabilities *icapabilities.ImageCapabilities) bool {
+	for _, capability := range capabilities.CapabilityVersions {
+		if capability.Capability.Name == "telegraf" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func processMonitoringClientCreationErr(diag *diag.Diagnostics, err error) {
+	if errutil.IsNotFound(err) {
```

Comment:
No reason, you've already added this processing in clients creating functions, haven't you?

Reply:
I didn't process error from 
```go
if err := monitoring.ListUsers(client, projectID).Err; err != nil {
	return nil, err
}
```
I should catch this error here

Reply:
Why should you? It would be better to process it in the corresponding methods of clients "configurer" structure.

Reply:
done